RELEASING.md

```diff
@@ -0,0 +1,120 @@
+The release process is pretty straightforward:
+
+## Pick a version number
+
+The release number uses a `major.minor` version format, and should
+just be numbers. If there's a major feature or a breaking change,
+don't be shy about bumping the major number: lots of people have a
+strong belief in SemVer and it would be a pity to let that faith down.
+
+## Set up a tracking issue in GitHub
+
+This isn't strictly necessary, but it's polite since it allows us to
+coordinate the work that we need to complete for the next release.
+
+## Make sure all dependencies are up to date
+
+Open up `repositories.bzl` and make sure that our dependencies are the
+most recent versions. If they're not, create a PR to update them, and
+land the change.
+
+## Update the `MODULE.bazel` file to reflect the latest changes
+
+There are three things you need to check
+
+1. The `version` parameter in the `module` declaration must be the
+   same as the version number of the release you're about to push.
+2. The `bazel_dep` entries must match the entries in
+   `repositories.bzl`
+3. The deps in the `maven.install` (named `rules_jvm_external_deps`)
+   must match the same deps from `repositories.bzl`
+
+You may well need to create a PR to update these too.
+
+## Tag the release in `git` and push the tag
+
+The tag format is just the release number (eg. `10.5`) without any
+additional prefixes or suffixes. We use a `major.minor` version
+format.
+
+```shell
+git switch master
+git pull
+git tag XX.YY
+git push --tags
+```
+
+## Create a draft entry in the GitHub releases.
+
+I like to copy the previous release, and then edit it. There's a handy
+button in the UI that will generate some release notes. Use it, then
+edit the values it gives you. The release notes should include user
+visible changes and all commits by people who aren't core
+committers. Bug fixes and little tweaks the core committers have made
+can be omitted if you don't think they're particularly noteworthy.
+
+Name the draft release after the version you've picked.
+
+## Create an archive of the release, and add as an asset
+
+GitHub periodically break how archives are created. To avoid this,
```

Comment:
I believe the URLs are different for manually created assets, right? Should we also update our default WORKSPACE snippets?

Reply:
That's what I did for the 5.0 release. If people copy-and-paste from an earlier release, then they'll be mostly there already :)